### PR TITLE
Make responseError compliant with exception's contract

### PR DESCRIPTION
**Pull request by ghostbuster91**
Before submitting pull request:
- [X] Check if the project compiles by running `sbt compile`
- [X] Verify docs compilation by running `sbt compileDocs`
- [X] Check if tests pass by running `sbt test`
- [X] Format code by running `sbt scalafmt`

This fixes https://gitter.im/softwaremill/sttp?at=5eff9841ec4a341bee067414

### CODE REVIEW

File: core/src/main/scala/sttp/client/ResponseAs.scala [hunk 1]
```diff
@@ -134,16 +136,25 @@ object ResponseAs {
...
     s =>
       doDeserialize(s) match {
         case Left(e)  => throw DeserializationError(s, e)
         case Right(b) => b
       }
 }
 
-sealed abstract class ResponseError[+T] extends Exception {
-  def body: String
+sealed abstract class ResponseError[+T](body: String) extends Exception(body)
```
Comment (adamw) [inline core/src/main/scala/sttp/client/ResponseAs.scala:147]: why would the (invalid) body be the exception's message?

File: core/src/main/scala/sttp/client/ResponseAs.scala [hunk 2]
```diff
@@ -134,16 +136,25 @@ object ResponseAs {
...
+case class HttpError(body: String) extends ResponseError[Nothing](body)
```
Comment (adamw) [inline core/src/main/scala/sttp/client/ResponseAs.scala:148]: that should probably mention the status code, if we want have an informative message

File: json/circe/src/main/scala/sttp/client/circe/SttpCirceApi.scala [hunk 3]
```diff
@@ -8,6 +9,10 @@ import sttp.client.{IsOption, ResponseAs, ResponseError}
 import sttp.model.MediaType
 
 trait SttpCirceApi {
+  implicit val errorMessageForCirceError: ErrorMessage[io.circe.Error] = new ErrorMessage[circe.Error] {
+    override def extract(t: circe.Error): String = t.getMessage
```
Comment (adamw) [inline json/circe/src/main/scala/sttp/client/circe/SttpCirceApi.scala:13]: that's simply the default, so we don't really need it?

File: core/src/main/scala/sttp/client/ResponseAs.scala [hunk 4]
```diff
@@ -134,16 +136,25 @@ object ResponseAs {
...
+case class DeserializationError[T: ErrorMessage](body: String, error: T)
+    extends ResponseError[T](implicitly[ErrorMessage[T]].extract(error))
+
+trait ErrorMessage[T] {
```
Comment (adamw) [inline core/src/main/scala/sttp/client/ResponseAs.scala:152]: maybe: `ShowError`?

File: json/circe/src/main/scala/sttp/client/circe/SttpCirceApi.scala (hunk 3 above)
Comment (ghostbuster91) [reply json/circe/src/main/scala/sttp/client/circe/SttpCirceApi.scala:13]: Good point, removing it...

File: core/src/main/scala/sttp/client/ResponseAs.scala (hunk 1 above)
Comment (ghostbuster91) [reply core/src/main/scala/sttp/client/ResponseAs.scala:147]: that's not a body - incorrect variable name

**Comment by adamw** [comment]
Released in 2.2.2, thanks!

**Pull request merged**